slixmpp/xmlstream/stanzabase.py

```diff
@@ -1037,8 +1038,11 @@ def _del_sub(self, name, all=False, lang=None):
             parent_path = "/".join(path[:len(path) - level - 1])
 
             elements = self.xml.findall(element_path)
-            parent = self.xml.find(parent_path)
-
+            try:
+                parent = self.xml.find(parent_path)
+            except:
```

Comment:
The exception catching should be specific rather than broad (and if broad, it should be logged at least in debug to make the information available)